PSCompatibilityCollector/Microsoft.PowerShell.CrossCompatibility/Commands/NewPSCompatibilityProfileCommand.cs

```diff
@@ -0,0 +1,182 @@
+        /// <value></value>
+        [Parameter]
+        public string PlatformId { get; set; }
+
+        /// <summary>
+        /// Sets the name of the profile file to be created in the default profile directory.
+        /// </summary>
+        [Parameter(ParameterSetName = "ProfileName")]
+        [ValidateNotNullOrEmpty]
+        public string ProfileName { get; set; }
+
+        /// <summary>
+        /// If set, validate the generated profile before returning it.
+        /// </summary>
+        [Parameter]
+        public SwitchParameter Validate { get; set; }
+
+        /// <summary>
+        /// Modules on paths starting with any of these will be excluded.
+        /// </summary>
+        /// <value></value>
+        [Parameter]
+        [ValidateNotNullOrEmpty]
+        public string[] ExcludeModulePathPrefix { get; set; }
+
+        /// <summary>
+        /// Assemblies on paths starting with any of these will be excluded.
+        /// </summary>
+        /// <value></value>
+        [Parameter]
+        [ValidateNotNullOrEmpty]
+        public string[] ExcludeAssemblyPathPrefix { get; set; }
+
+        protected override void EndProcessing()
+        {
+            CompatibilityProfileData profile;
+            IEnumerable<Exception> errors;
+            using (var pwsh = SMA.PowerShell.Create())
+            {
+                var collectorBuilder = new CompatibilityProfileCollector.Builder();
+
+                if (ExcludeModulePathPrefix != null && ExcludeModulePathPrefix.Length > 0)
+                {
+                    collectorBuilder.ExcludedModulePathPrefixes = ExcludeModulePathPrefix;
+                }
+
+                if (ExcludeAssemblyPathPrefix != null && ExcludeAssemblyPathPrefix.Length > 0)
+                {
+                    collectorBuilder.ExcludedAssemblyPathPrefixes = ExcludeAssemblyPathPrefix;
+                }
+
+                using (var profileCollector = collectorBuilder.Build(pwsh))
+                {
+                    profile = string.IsNullOrEmpty(PlatformId)
+                        ? profileCollector.GetCompatibilityData(out errors)
+                        : profileCollector.GetCompatibilityData(PlatformId, out errors);
+                }
+            }
+
+            // Report any problems we hit
+            foreach (Exception e in errors)
+            {
+                this.WriteExceptionAsWarning(e);
+            }
+
+            if (Validate)
+            {
+                var validator = new QuickCheckValidator();
+                if (validator.IsProfileValid(profile, out IEnumerable<Exception> validationErrors))
+                {
+                    foreach (Exception validationError in validationErrors)
+                    {
+                        this.WriteExceptionAsWarning(validationError);
+                    }
+                }
+            }
+
+            // If PassThru is set, just pass the object back and we're done
+            if (PassThru)
```

Comment:
`PassThru` is usually used more like `/bin/tee` where the data _is_ returned _and_ the targets are created.

Reply:
Yeah I was thinking about that. Maybe I should change this parameter name? Nothing else common really captures the behaviour though

Reply:
yah, probably, but later before we release and have time to think of a better parameter name